File: コードレビュー/05.マスター/プログラム作成問題：マスター編/20200127_165353689/RandomGenerator.java
import java.util.Random;

public class RandomGenerator {

	/**
	 * 最小値と最大値を渡してその範囲内のランダムな数字を返す
	 * @param min
	 * @param max
	 * @return
	 */
	public static int RandomRange(int min, int max) {
		Random random = new Random();
		return random.nextInt(max - min) + min;
	}

	/**
	 * 成功確率のパーセンテージを渡して成功したかどうかを判定する
	 * @param probability
	 * @return
	 */
	public static boolean RandomJudge(int probability) {
		int judgementValue = RandomRange(0, 100);

		if (probability >= judgementValue) {
			return true;
		}
		return false;

	}
}

File: コードレビュー/07.Unityアプリ/20200701_USSBB1M39/Assets/Script/RandomGenerator.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class RandomGenerator : MonoBehaviour
{
	/**
	 * 成功確率のパーセンテージを渡して成功したかどうかを判定する
	 * @param probability
	 * @return
	 */
	public static bool RandomJudge(int probability)
	{
		int judgementValue = UnityEngine.Random.Range(0, 100);

		if (probability >= judgementValue)
		{
			return true;
		}
		return false;

	}
}
